src/components/LoginPanel.tsx

```diff
@@ -37,8 +37,14 @@ const LoginPanel: React.FC<{}> = () => {
     const supabase = createClientComponentClient<DatabaseAutoType>();
     const theme = useTheme();
 
+    const [loaded, setLoaded] = React.useState(false);
+
+    useEffect(() => {
+        setLoaded(true);
```

Comment:
Why do we need this? Is it different from just setting `data-loaded={true}`?

Reply:
data-loaded={true} would mean on a server-side-rendered document, it would still be true. The key thing here is that `useEffect` only runs after hydration, so `loaded === true` implies that hydration has happened and themeing + routing infrastructure should be in place.